### PR TITLE
Add base tests for 5 utility functions

### PR DESCRIPTION
**Pull request by II-KA**

### CODE REVIEW

**Comment by II-KA** [comment]
Missing tests for memoize.js, but feel free to begin reviewing the existing tests

File: test/filter.test.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    });
+    it('there is no users', () => {
+      const result = filter([], active);
+      /* Expecting an empty array, since there is no users */
+      expect(result).toBeArrayOfSize(0);
+    });
+    it('users array is null', () => {
+      const result = filter(null, active);
+      /* Expecting an empty array, since array is null */
+      expect(result).toBeArrayOfSize(0);
```
Comment (II-KA) [inline test/filter.test.js]: These fail, returning `[[]]` instead of `[]`. I'd expect filter to return an empty array (as [js filter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) does) if none of the values match the predicate. Do you agree @Liljenmaa?

File: test/defaultToAny.test.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+      expect(result).toBeArrayOfSize(0);
+    });
+    it('value is null', () => {
+      const result = defaultToAny(null);
+      expect(result).toBeArrayOfSize(0);
+    });
+    it('value is undefined', () => {
+      const result = defaultToAny(undefined);
+      expect(result).toBeArrayOfSize(0);
+    });
```
Comment (II-KA) [inline test/defaultToAny.test.js]: These return themselves, which is a bit unexpected. This is a corner case though and I'm not sure what the correct return value should be. I went with an empty array, because that's what defaultValues are in `defaultToAny.js` in this case. Thoughts?

**Comment by II-KA** [comment]
`defaultTo.js` and `defaultToAny.js` do not handle NaN values correctly due to using `==` instead of `===`

**Comment by II-KA** [comment]
Added base tests for `memoize.js`. Missing coverage for lines 55-58, not sure why

**Comment by II-KA** [comment]
Removed the dummy tests, added more tests for filter.js

**Comment by II-KA** [comment]
Now test coverage is 100% for each tested utility function

File: test/filter.test.js (hunk 1 above)
Comment (Liljenmaa) [reply test/filter.test.js]: Yes

File: test/defaultToAny.test.js (hunk 2 above)
Comment (Liljenmaa) [reply test/defaultToAny.test.js]: You're not giving any default values, which would suggest this is undefined behaviour in which case it should throw an IllegalArgumentException or similar
